scripts/performance-tests-mx.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+# Disable the Python keyring, otherwise poetry sometimes asks for password. See
+#  https://github.com/pypa/pip/issues/7883
+export PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring
+
+MX_VERSION=${MX_VERSION:-'master'}
+
+SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
+
+MASTER_COMMIT="$(git rev-parse origin/main)"
+MASTER_COMMIT_SHORT="$(git rev-parse --short origin/main)"
+
+FEATURE_BRANCH_NAME=${FEATURE_BRANCH_NAME:-"$(git rev-parse --abbrev-ref HEAD)"}
+FEATURE_BRANCH_NAME="${FEATURE_BRANCH_NAME//\//-}"
+
+PYTEST_PARALLEL=${PYTEST_PARALLEL:-3}
+
+if [[ $FEATURE_BRANCH_NAME == "master" ]]; then
+  FEATURE_BRANCH_NAME="feature"
+fi
+
+# Create a temporary directory and store its name in a variable.
+TEMPD=$(mktemp -d)
+
+# Exit if the temp directory wasn't created successfully.
+if [ ! -e "$TEMPD" ]; then
+    >&2 echo "Failed to create temp directory"
+    exit 1
+fi
+
+# Make sure the temp directory gets removed and kore-rpc-booster gets killed on script exit.
+trap "exit 1"           HUP INT PIPE QUIT TERM
+trap 'rm -rf "$TEMPD" && killall kore-rpc-booster || echo "No zombie processes found"'  EXIT
+
+feature_shell() {
+  GC_DONT_GC=1 nix develop . --extra-experimental-features 'nix-command flakes' --override-input k-framework/booster-backend $SCRIPT_DIR/../ --command bash -c "$1"
+}
+
+master_shell() {
+  GC_DONT_GC=1 nix develop . --extra-experimental-features 'nix-command flakes' --override-input k-framework/booster-backend github:runtimeverification/hs-backend-booster/$MASTER_COMMIT --command bash -c "$1"
+}
+
+cd $TEMPD
+git clone --depth 1 --branch $MX_VERSION https://github.com/runtimeverification/mx-backend.git
+cd mx-backend
+
+if [[ $MX_VERSION == "master" ]]; then
+  MX_VERSION=$(git name-rev --tags --name-only $(git rev-parse HEAD))
+else
+  MX_VERSION="${MX_VERSION//\//-}"
+fi
+
+git submodule update --init --recursive --depth 1 kmxwasm/k-src/mx-semantics/
+
+
+feature_shell "poetry -C kmxwasm install && make -C kmxwasm kbuild-haskell"
```

Comment:
Do we need to execute the same command in `master_shell`? 

Also, will the LLVM backend shared library be built by `make -C kmxwasm kbuild-haskell`? @virgil-serbanuta @bbyalcinkaya could you have a look at this PR and make sure it does what you'd expect?

Reply:
yes, master shell is used fto compare our feature branch change to the current version of booster in mx

Reply:
these steps are copied from the CI run in the mx repo:
https://github.com/runtimeverification/mx-backend/blob/872626e6a658bdbdc7914e5c1c01d19cffc3f11c/.github/workflows/test.yml#L96-L118
so hopefully should do the same thing....

Reply:
I would _guess_ that the `poetry install` and `make kbuild-haskell` are independent of the booster version, so we wouldn't  have to repeat that command (we don't at the moment). 
Better to have this confirmed by @virgil-serbanuta  or @bbyalcinkaya , though.

Reply:
The kasmer tool builds the semantics at start (for the Haskell backend, for the LLVM backend, and, if `--booster` is passed to the tool, it also builds the llvm library). Tests do something similar. `make -C kmxwasm kbuild-haskell` should not be needed unless you need to run things outside our tool (or, perhaps, you just want to check that it compiles).

Reply:
FWIW, we will probably add the option to use an already-built semantics soon, we can try to do that sooner if you need it.

Reply:
![image](https://github.com/runtimeverification/hs-backend-booster/assets/669076/ad0afc6f-3862-4598-b830-ae066c0158d2)